### PR TITLE
Implement reading an object from the object database.

### PR DESCRIPTION
**Pull request by kaendfinger**
I've been using Rust for only two days, so feel free to criticize everything :)

### CODE REVIEW

**Comment by kaendfinger** [comment]
Related to #124, however this is a quite different API.
I'll look into a writing API too, but reading is much more important to have this level of an API, I feel.

File: src/odb.rs [hunk 1]
```diff
@@ -66,6 +68,70 @@ impl<'repo> Odb<'repo> {
...
+    pub fn read(&self, oid: Oid) -> Result<OdbObject, Error> {
+        let mut out = ptr::null_mut();
+        unsafe {
+            try_call!(raw::git_odb_read(&mut out, self.raw, oid.raw()));
+            Ok(OdbObject::from_raw(out))
+        }
+    }
+
+    /// Checks if the object database has an object.
+    pub fn exists(&self, oid: Oid) -> Result<bool, Error> {
```
Comment (alexcrichton) [inline src/odb.rs]: Should this perhaps just return a `bool` instead of a `Result`? (not sure it can return an `Error`

File: src/odb.rs [hunk 2]
```diff
@@ -66,6 +68,70 @@ impl<'repo> Odb<'repo> {
...
+
+impl<'a> Drop for OdbObject<'a> {
+    fn drop(&mut self) {
+        unsafe { raw::git_odb_object_free(self.raw) }
+    }
+}
+
+impl<'a> OdbObject<'a> {
+    /// Get the object type.
+    pub fn get_type(&self) -> Result<ObjectType, Error> {
```
Comment (alexcrichton) [inline src/odb.rs]: Similar to above (and the ones below) I don't think a `Result` needs to be returned here, right?

File: src/odb.rs (hunk 1 above)
Comment (kaendfinger) [reply src/odb.rs]: That's correct, I'll fix that.

File: src/odb.rs (hunk 2 above)
Comment (kaendfinger) [reply src/odb.rs]: I have fixed this, but it requires calling unwrap(), which feels wrong in a library, let me know what you think.

**Comment by alexcrichton** [comment]
Thanks!

**Pull request merged**